lib/cli/parse-args.js

```diff
@@ -0,0 +1,126 @@
+// TODO: Consider publishing as independent package
+
+'use strict';
+
+const ServerlessError = require('../serverless-error');
+
+const paramRe = /^-(?:-(?<fullName>[a-z][A-Za-z0-9:-]+)|(?<aliasNames>[a-z]+))(?:=(?<value>.+)|$)/;
+const isParamName = RegExp.prototype.test.bind(paramRe);
+
+module.exports = (
+  args,
+  { boolean = new Set(), alias = new Map(), string = new Set(), multiple = new Set() }
+) => {
+  const result = Object.create(null);
+  result._ = [];
+  for (let i = 0, arg; (arg = args[i]); ++i) {
+    if (arg === '--') {
+      result._.push(...args.slice(i + 1));
+      break;
+    }
+    const paramMatch = arg.match(paramRe);
+    if (!paramMatch) {
+      result._.push(arg);
+      continue;
+    }
+    const { fullName, aliasNames, value } = paramMatch.groups;
+    const isBoolean = !value && (!args[i + 1] || args[i + 1] === '--' || isParamName(args[i + 1]));
+
+    let paramName;
+    if (aliasNames) {
+      if (aliasNames.length > 1) {
+        if (value) {
+          throw new ServerlessError(
+            `Unexpected value for "--${aliasNames}"`,
```

Comment:
nit: should it error out with only one dash e.g `-${aliasNames}` here? Also, why only throw when aliasNames is longer than 1 - is it to support such scenario:
- Someone passes e.g. `-ab` which should be recognized as both `-a` and `-b` flag, so if in that case there's also a value, we have no idea for which of these it should apply?

Reply:
> nit: should it error out with only one dash e.g -${aliasNames} here?

Good point! Fixed locally

> Someone passes e.g. -ab which should be recognized as both -a and -b flag, so if in that case there's also a value, we have no idea for which of these it should apply?

Yes, if it's `-ab`, internal assumption is that it can be only about boolean (value-less) params.

Reply:
Thanks for clarification :bow: